src/optimize/dynamic_dll_plugin/dll_config_model.js

```diff
@@ -0,0 +1,242 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { fromRoot, IS_KIBANA_DISTRIBUTABLE } from '../../utils';
+import webpack from 'webpack';
+import webpackMerge from 'webpack-merge';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+import UglifyJsPlugin from 'uglifyjs-webpack-plugin';
+
+function generateDLL(config) {
+  const {
+    dllAlias,
+    dllNoParseRules,
+    dllContext,
+    dllEntry,
+    dllOutputPath,
+    dllPublicPath,
+    dllBundleName,
+    dllBundleFilename,
+    dllStyleFilename,
+    dllManifestPath
+  } = config;
+
+  const BABEL_PRESET_PATH = require.resolve('@kbn/babel-preset/webpack_preset');
+  const BABEL_EXCLUDE_RE = [
+    /[\/\\](webpackShims|node_modules|bower_components)[\/\\]/,
+  ];
+
+  return {
+    entry: dllEntry,
+    context: dllContext,
+    output: {
+      filename: dllBundleFilename,
+      path: dllOutputPath,
+      publicPath: dllPublicPath,
+      library: dllBundleName
+    },
+    node: { fs: 'empty', child_process: 'empty', dns: 'empty', net: 'empty', tls: 'empty' },
+    resolve: {
+      extensions: ['.js', '.json'],
+      mainFields: ['browser', 'browserify', 'main'],
+      alias: dllAlias,
+      modules: [
+        'webpackShims',
+        fromRoot('webpackShims'),
+
+        'node_modules',
+        fromRoot('node_modules'),
+      ],
+    },
+    module: {
```

Comment:
There is a lot duplicated in here from the optimizer, is it possible to reference the configuration in the optimizer here?

Reply:
In the first place I was passing the original base optimizer config object and picking some of the configs from there. But then @spalger and I agreed that the base_optimizer config doesn't change so much and it's more readable to have the plain config here. However we could always change this in the future. We just need to pick more config into the dynamic_dll_plugin from the base_optimizer and pass them to the dll config model.